src/indexing.hpp

```diff
@@ -151,6 +151,55 @@ std::string getIndexString( INDEX const index, REMAINING_INDICES const ... indic
   return oss.str();
 }
 
+/**
+ * @tparam INDEX_TYPE The integral type to be printed.
+ * @param val the value to be printed
+ */
+template< typename INDEX_TYPE >
+LVARRAY_HOST_DEVICE
+void printIndexValue( INDEX_TYPE const & val )
+{
+  using DecayedType = std::decay_t<INDEX_TYPE>;
+  static_assert( std::is_integral_v< DecayedType >, "INDEX_TYPE must be an integral type." );
```

Comment:
either I have found compiler bug or something really weird is happening coz I keep getting this error on one of the CI jobs:
https://github.com/GEOS-DEV/GEOS/actions/runs/12938898965/job/36090106173 
I don't see how that index could be anything other than an `integral_type`.

Reply:
Removing the assert and forcing a `static_cast` to a` long long` works. However, I don't know what non integral type the compiler thinks that index is since it can cast it to a long long without any issue. 
I think we can leave it as is for now.